File: Larchik.Domain/Deal.cs
namespace Larchik.Domain;

public class Deal
{
    public Guid Id { get; set; }
    public int Quantity { get; set; }
    public double Price { get; set; }
    public double Amount { get; set; }
    public Operation? Operation { get; set; }
    public string OperationId { get; set; } = string.Empty;
    public Stock Stock { get; set; } = null!;
    public string StockId { get; set; } = string.Empty;
    public double Commission { get; set; }
    public DateTime CreatedAt{ get; set; }
    public Account Account { get; set; } = null!;
    public Guid AccountId { get; set; }
}

File: client-app/src/app/models/deal.ts
export interface Deal {
    id: string;
    quantity: number;
    price: number;
    type: number | undefined;
    currency: string;
    stock: string | undefined;
    commission: number;
    createdAt: Date;
    accountId: string;
}

export class Deal implements Deal {
    constructor(init?: DealFormValues) {
      Object.assign(this, init);
    }
  }

export class DealFormValues {
    id: string = '';
    quantity: number = 1;
    price: number = 0;
    type: number | undefined = undefined;
    currency: string = '';
    stock: string | undefined;
    commission: number = 0;
    createdAt: Date = new Date();
    accountId: string = '';

    constructor(deal?: DealFormValues) {
        if (deal) {
            this.id = deal.id;
            this.quantity = deal.quantity;
            this.price = deal.price;
            this.type = deal.type;
            this.currency = deal.currency;
            this.stock = deal.stock;
            this.commission = deal.commission;
            this.createdAt = deal.createdAt;
            this.accountId = deal.accountId;
        }
    }
}

export enum DealKind {
    Add = 1,
    Withdrawal,
    Purchase,
    Sale,
    Commission,
    Tax,
    Dividends
}